pkg/agent/proxy/proxier.go

```diff
@@ -1099,18 +1128,70 @@ func (p *proxier) OnEndpointSlicesSynced() {
 	}
 }
 
+func (p *proxier) matchAddressFamily(eps *discovery.EndpointSlice) bool {
+	switch eps.AddressType {
+	case discovery.AddressTypeIPv4:
+		return !p.isIPv6
+	case discovery.AddressTypeIPv6:
+		return p.isIPv6
+	default:
+		return false
+	}
+}
+
 func (p *proxier) OnServiceAdd(service *corev1.Service) {
 	klog.V(2).InfoS("Processing Service ADD event", "Service", klog.KObj(service))
 	p.OnServiceUpdate(nil, service)
 }
 
+func getServiceIPFamilySupport(preSvc, curSvc *corev1.Service) (bool, bool) {
```

Comment:
I don't think there is much value in considering `preSvc` in this function?

Should the implementation use `utilproxy.GetClusterIPByFamily`?